tools/c7n_azure/c7n_azure/utils.py

```diff
@@ -99,6 +100,41 @@ def azure_name_value_pair(name, value):
     return NameValuePair(**{'name': name, 'value': value})
 
 
+send_logger = logging.getLogger('custodian.azure.utils.ServiceClient.send')
+
+
+def custodian_azure_send_override(self, request, headers=None, content=None, **kwargs):
+    """ Overrides ServiceClient.send() function to implement retries & log headers
+    """
+    retries = 0
+    max_retries = 3
+    while retries < max_retries:
+        response = self.old_send(request, headers, content, **kwargs)
+
+        send_logger.debug(response.status_code)
+        for k, v in response.headers.items():
+            if k.startswith('x-ms-ratelimit'):
+                send_logger.debug(k + ':' + v)
+
+        # Retry codes from urllib3/util/retry.py
+        if response.status_code in [413, 429, 503]:
+            retry_after = None
+            for k in response.headers.keys():
+                if StringUtils.equal('retry-after', k):
+                    retry_after = int(response.headers[k])
+
+            if retry_after is not None and retry_after < 30:
```

Comment:
(minor) lift the 30 seconds up to a var, so we can parameterize this in the future easier.